lib/facter/iis_version.rb

```diff
@@ -1,20 +1,15 @@
-# iis_version.rb
-Facter.add("iis_version") do
- confine :kernel => :windows
+Facter.add(:iis_version) do
+  confine :kernel => :windows
   setcode do
-    begin
-      psexec = if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
-                 "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
-               elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
-                "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
-               else
-                'powershell.exe'
-               end
-      iis_ver = %x{#{psexec} -ExecutionPolicy ByPass -Command "$regkey = Get-ItemProperty HKLM:\\SOFTWARE\\Microsoft\\InetStp\\ -Name VersionString -ea silentlycontinue;if ($regkey) {$regkey.VersionString.SubString(8,3)}"}
-    rescue
-      iis_ver = ""
-    end
-
-    iis_ver
-  end
+	version = nil
+	require 'win32/registry'
```

Comment:
Hi Igor,

To the best of my knowledge it is included in the redistributed Ruby that comes at least with agent 3.8.4 for Windows, I verified that by quickly setting up a lab environment with a clean puppet master and a clean Server 2012 r2 Windows box and it worked out of the box. I've tested it in our test environment to, but those to are on 3.8.4 although the veriety of Windows versions are greater, but so are the modules installed on that puppet master. Do you want me to test with older versions of agents to verify it further? I've tried to verify when exactly the gem registry.rb showed up in puppet for windows but haven't found anything on that.

Best,
Fredrik

Den 02 dec 2015 kl. 15:00 skrev Igor Galić <notifications@github.com>:

In lib/facter/iis_version.rb:
> -      psexec = if File.exists?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
> -                 "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
> -               elsif File.exists?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
> -                "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
> -               else
> -                'powershell.exe'
> -               end
> -      iis_ver = %x{#{psexec} -ExecutionPolicy ByPass -Command "$regkey = Get-ItemProperty HKLM:\\SOFTWARE\\Microsoft\\InetStp\\ -Name VersionString -ea silentlycontinue;if ($regkey) {$regkey.VersionString.SubString(8,3)}"}
> -    rescue
> -      iis_ver = ""
> -    end
> -
> -    iis_ver
> -  end
> +	version = nil
> +	require 'win32/registry'
is this gem installed by default on windows?
—
Reply to this email directly or view it on GitHub.